site/app/views/forum/ForumThreadView.php

```diff
@@ -590,162 +590,150 @@ public function showAlteredDisplayList($threads, $filtering, $thread_id, $catego
-								$fa_font_size = "1.5em";
-								$tooltip = "Thread Resolved";
-							} else {
-								$fa_icon = "fa-question";
-								$fa_color = "#ffcc00";
-								$fa_margin_right = "5px";
-								$fa_font_size = "1.8em";
-								$tooltip = "Thread Unresolved";
-							}
-							$return .= <<<HTML
-							<i class="fa ${fa_icon}" style="margin-right:${fa_margin_right}; padding-left:3px; position:relative; float:right; display:inline-block; color:${fa_color}; font-size:${fa_font_size};" title = "${tooltip}" aria-hidden="true"></i>
-HTML;
-						}
-						if($this->core->getQueries()->isThreadLocked($thread['id']))
-                        {
-                            $return .= <<<HTML
-                            <i class="fas fa-lock" style="padding-left:3px;position:relative; float:right; display:inline-block; color: white; -webkit-text-stroke-width: 1px;
-    -webkit-text-stroke-color: black;" title="Locked" aria-hidden="true"></i>
-HTML;
+            $sizeOfContent = strlen($first_post_content);
+            $contentDisplay = substr($first_post_content, 0, ($sizeOfContent < 80) ? $sizeOfContent : strrpos(substr($first_post_content, 0, 80), " "));
+            $titleLength = strlen($thread['title']);
 
-                        }
-						$categories_content = array();
-						foreach ($thread["categories_desc"] as $category_desc) {
-							$categories_content[] = array(htmlentities($category_desc, ENT_QUOTES | ENT_HTML5, 'UTF-8'));
-						}
-						for ($i = 0; $i < count($thread["categories_color"]); $i+=1) {
-							$categories_content[$i][] = $thread["categories_color"][$i];
-						}
-						$return .= <<<HTML
-						<h4>{$titleDisplay}</h4>
-						<h5 style="font-weight: normal;">{$contentDisplay}</h5>
-HTML;
-						foreach ($categories_content as $category_content) {
-							$return .= <<<HTML
-							<span class="label_forum" style="background-color: {$category_content[1]}">{$category_content[0]}</span>
-HTML;
-						}
-						if(!is_null($date)) {
-							$return .= <<<HTML
-							<h5 style="float:right; font-weight:normal;margin-top:5px">{$function_date($date,"n/j g:i A")}</h5>
-HTML;
-						}
-						$return .= <<<HTML
-						</div>
-						</a>
-						<hr style="margin-top: 0px;margin-bottom:0px;">
-HTML;
-					}
-					return $return;
-	}
+            $titleDisplay = substr($titleDisplay, 0, ($titleLength < 40) ? $titleLength : strrpos(substr($titleDisplay, 0, 40), " "));
+
+            if (strlen($first_post["content"]) > 80) {
+                $contentDisplay .= "...";
+            }
+            if (strlen($thread["title"]) > 40) {
+                //Fix ... appearing
+                if (empty($titleDisplay))
+                    $titleDisplay .= substr($thread['title'], 0, 30);
+                $titleDisplay .= "...";
+            }
+            $titleDisplay = ($display_thread_ids ? "({$thread['id']}) " : '') . $titleDisplay;
+            $titleDisplay = htmlentities($titleDisplay, ENT_QUOTES | ENT_HTML5, 'UTF-8');
```

Comment:
As @MasterOdin mentioned in #3768, we don't have to include `htmlentities` as twig will take care of this for us.